harness/determined/pytorch/_pytorch_context.py

```diff
@@ -189,12 +189,19 @@ def wrap_lr_scheduler(
         The LR scheduler must use an optimizer wrapped by :meth:`wrap_optimizer`.  If ``apex.amp``
         is in use, the optimizer must also have been configured with :meth:`configure_apex_amp`.
         """
+        opt = getattr(lr_scheduler, "optimizer", None)
+        if opt is not None:
+            check.is_in(
+                opt,
+                self.optimizers,
+                "Must use an optimizer that is returned by wrap_optimizer()",
 
-        check.is_in(
-            lr_scheduler.optimizer,  # type: ignore
-            self.optimizers,
-            "Must use an optimizer that is returned by wrap_optimizer()",
-        )
+            )
+        else:
+            logging.warning(
+                "You are using custom LR scheduler, please remember to use the wrapped"
+                "optimizer returned by wrap_optimizer."
+            )
```

Comment:
Can we avoid emitting "just in case" warnings when we don't know that something is wrong?